src/main/scala/it/agilelab/gis/core/model/geometry/Circle.scala

```
@@ -18,11 +16,37 @@ object Circle {
     val fact = new GeometryFactory
     val coordinate = new Coordinate(mbr.getMinX + radius, mbr.getMinY + radius)
     val point = fact.createPoint(coordinate)
-    new Circle(point, radius)
+    val factory = new GeometryFactoryEnriched()
+    new Circle(point, radius, factory)
   }
 }
 
-case class Circle(center: Point, radius: Double) extends Serializable {
+/** Class extends Geometry and represent a Circle
+  *
+  * @param center a Point representing the center of the circle
+  * @param radius Double representing the radius of the circle
+  * @param circleFactory a GeometryFactory needed to create the geometry
+  */
+case class Circle(center: Point, radius: Double, circleFactory: GeometryFactoryEnriched)
+    extends Geometry(circleFactory: GeometryFactoryEnriched) {
+
+  private lazy val coordinates = getCoordinates()
+  protected lazy val shell: LinearRing = factory.createLinearRing(coordinates)
+
+  /** Define a shell: a square (built as LinearRing) that contains the circle
+    */
+  override def getCoordinates(): Array[Coordinate] = {
+    val coordinates_0 = new Coordinate(getMBR.getMinX, getMBR.getMinY)
+    val coordinates_1 = new Coordinate(getMBR.getMaxX, getMBR.getMinY)
+    val coordinates_2 = new Coordinate(getMBR.getMaxX, getMBR.getMaxY)
+    val coordinates_3 = new Coordinate(getMBR.getMinX, getMBR.getMaxY)
+    Array[Coordinate](coordinates_0, coordinates_1, coordinates_2, coordinates_3, coordinates_0)
+  }
+
+  def apply(geomComponentFilter: GeometryComponentFilter): Unit = ???
```

we could throw this exception in these cases NotImplementedException